src/components/Expenses/ExpensesAnalysisChart.jsx

```diff
@@ -0,0 +1,174 @@
+//React import
+import React, { useState, useEffect } from "react";
+// React-chart-js-2 import
+import { Bar } from "react-chartjs-2";
+// Auth0 import
+import { useAuth0 } from "@auth0/auth0-react";
+
+export function ExpensesAnalysis() {
+  // Auth0 Hooks
+  const { getAccessTokenSilently } = useAuth0();
+  //  React hooks
+  const [expenseData, setExpenseData] = useState([]);
+  const [yearOfExpense, setYearOfExpense] = useState([]);
+  const [selectedYear, setSelectedYear] = useState(`${new Date().getFullYear()}`); // Initial state of the current year
+  // Constants
+  const yearArray = [];
+  const labels = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec",];
+  const colourForBar = ["rgba(46, 134, 193, 0.6)", "rgba(211, 84, 0, 0.6)"];
+
+  // useEffect to re-render based on selectedYears state change
+  useEffect(() => {
```

Comment:
You have basically the same fetch for expenses in the expense component too. Maybe they could be passed to a parent component and then set the state for both of them from just one fetch?